src/Sql/dbo/Functions/PolicyApplicableToUser.sql

```diff
@@ -25,11 +25,12 @@ LEFT JOIN
 WHERE
     (
         (
-            OU.[Status] > 0
+            OU.[UserId] IS NOT NULL     -- OrgUsers who have accepted their invite and are linked to a UserId
             AND OU.[UserId] = @UserId 
         )
         OR (
-            OU.[Status] = 0 -- 'Invited' OrgUsers are not linked to a UserId yet, so we have to look up their email
+            OU.[Email] IS NOT NULL      -- OrgUsers who have not accepted their invite are not linked to a UserId yet, 
```

Comment:
That also makes these checks more explicit on status and intent rather than inferred on a current side-effect of application behavior. I like explicit intent vs. relying on application behavior when it comes to data design

Reply:
> This is a redundant check, if OU.[Email] is NULL, it will fail the test below in the IN statement. What I am unsure of is whether this would improve query performance when the query optimizer looks at execution order for this check when it's on the same column, Email.

Good point. For what it's worth, I had a peek at the query execution plan here and it seems fairly terrible:
![Screen Shot 2022-08-01 at 2 00 51 pm](https://user-images.githubusercontent.com/31796059/182073317-a0fadeaa-24ef-4c10-b8b1-4c11efc50c27.png)

From what I can tell: this `EMAIL IS NOT NULL` / `Status == 0` check basically means that more unwanted rows are filtered out when the table is first scanned (far right-hand operation) instead of when the `User` and `OrganizationUser` tables are joined. Which seems a bit better to me, although overall it seems like we need to re-think the query as a whole.

Anyway, moving on to what we check here...

The risk of checking `Status` is that it won't return anything for a user who is invited but then revoked. (Their status will not be 0, so it will try to match them by `UserId`, but they're not linked to a `UserId` yet.) That's not a problem right now, but I thought it was an undocumented assumption that might cause regressions in the future. I'm on the fence about it though, so I think you're right that it's better to be explicit.

Reply:
If you're correct, then let's keep what you have I suppose, either way we need to perform some filtering before the join where we can. Overall, if the performance of this sproc is poor and the query plan needs tuning, this may be something that needs to be addressed before it is released given how often this sproc will be executed for any and all org users and that could negatively impact performance.

Reply:
This sproc is already in production, we're just adjusting this conditional, which shouldn't have any real impact on performance. I'm also not 100% sure the query plans on my small dev db accurately reflect what's happening in prod - this is something I want to dig into further in my BEEEP time. But for now I think we can leave it - I just wanted an excuse to dig into it :D